ion-schema/src/type_reference.rs

```diff
@@ -0,0 +1,72 @@
+use crate::ion_path::IonPath;
+use crate::isl::isl_type_reference::IslTypeRefModifier;
+use crate::result::ValidationResult;
+use crate::system::{TypeId, TypeStore};
+use crate::types::TypeValidator;
+use crate::IonSchemaElement;
+use ion_rs::IonType;
+
+/// Provides an internal representation of a schema type reference.
```

Comment:
I meant to ISL type reference but I realized there is no definition of that in the spec. Does something like following sound right to you:
```suggestion
/// Provides reference to a type definition.
/// This will be used by constraints to store type references.
```